src/main/java/seedu/uninurse/model/task/TaskList.java

```diff
@@ -95,13 +103,13 @@ public boolean isEmpty() {
     /**
      * Returns the internal task list.
      */
-    public ArrayList<Task> getTasks() {
-        return internalTaskList;
+    public List<Task> getTasks() {
+        return filteredTaskLists;
     }
```

Comment:
I have changed the return type to List<Tasks>, but decided to keep the method getTasks() and instead use it to return filteredTaskList, specific to TaskList